src/protocols/radius/list.c

```diff
@@ -590,6 +593,18 @@ bool fr_packet_list_id_alloc(fr_packet_list_t *pl, int proto,
 		 *	Otherwise, this socket is OK to use.
 		 */
 
+		/*
+		 *	An explicit ID was requested
+		 */
+
+		if (id != -1) {
+			if  ((ps->id[(id >> 3) & 0x1f] & (1 << (id & 0x07))) != 0) continue;
```

Comment:
Hmm what's the idea here?

Reply:
The IDs are stored as a bitmask, so this is checking if the ID is allocated.